src/DataLib/fendl32B_retrofit/groupr_tools.py

```diff
@@ -110,16 +112,57 @@ def write_njoy_input_file(template):
     with open('groupr.inp', 'w') as f:
```

Comment:
noticing this filename is hardcoded in different places, so might want to define this in a single place and pass it around or assume it's global.

Also - this method may be too simple and clear to need a standalone method rather than just include the code where you need it.